rest/tests/test_rest.py

```diff
@@ -0,0 +1,742 @@
+            self.repo.slug, vocab.slug, input_dict
+        )
+        new_term_slug = output_dict['slug']
+
+        # patch with missing slug
+        self.patch_term(
+            self.repo.slug, vocab.slug, "missing", {'label': 'rename'},
+            expected_status=NOT_FOUND
+        )
+
+        output_dict = self.patch_term(
+            self.repo.slug, vocab.slug, new_term_slug, {'label': 'rename'}
+        )
+
+        new_term_slug = output_dict['slug']
+        self.assertEqual('rename',
+                         Term.objects.get(slug=new_term_slug).label)
+
+        # put with missing slug
+        self.put_term(
+            self.repo.slug, vocab.slug, "missing slug", input_dict,
+            expected_status=NOT_FOUND
+        )
+
+        output_dict = self.put_term(
+            self.repo.slug, vocab.slug, new_term_slug, input_dict
+        )
+        new_term_slug = output_dict['slug']
+        self.assertEqual(input_dict['label'],
+                         Term.objects.get(slug=new_term_slug).label)
+        # never created a duplicate
+        self.assertEqual(1, Term.objects.count())
+
+        # test missing repository
+        self.get_term(self.repo.slug, vocab.slug, "xyz",
+                      expected_status=NOT_FOUND)
+
+        vocab2 = create_vocabulary(
+            self.repo.id,
+            "other name",
+            "description",
+            True,
+            Vocabulary.FREE_TAGGING,
+            1000
+        )
+
+        self.delete_term(self.repo.slug, vocab.slug, new_term_slug)
+
+        # make sure terms only show up under vocabulary they belong to
+        term1 = self.create_term(self.repo.slug, vocab.slug)
+        term2 = self.create_term(self.repo.slug, vocab2.slug)
+        self.create_term(self.repo.slug, vocab.slug,
+                         expected_status=BAD_REQUEST)
+
+        terms = self.get_terms(self.repo.slug, vocab.slug)
+        self.assertEqual(1, terms['count'])
+        terms = self.get_terms(self.repo.slug, vocab2.slug)
+        self.assertEqual(1, terms['count'])
+
+        self.get_term(self.repo.slug, vocab.slug, term1['slug'])
+        self.get_term(self.repo.slug, vocab.slug, term2['slug'],
+                      expected_status=NOT_FOUND)
+        self.get_term(self.repo.slug, vocab2.slug, term1['slug'],
+                      expected_status=NOT_FOUND)
+        self.get_term(self.repo.slug, vocab2.slug, term2['slug'])
+
+    def test_vocabulary_pagination(self):
+        """Test pagination for collections"""
+
+        expected = [
+            create_vocabulary(
+                self.repo.id,
+                "name{i}".format(i=i),
+                "description",
+                True,
+                Vocabulary.FREE_TAGGING,
+                1000
+            ) for i in range(100)]
+
+        resp = self.client.get(
+            '/api/v1/repositories/{repo_slug}/vocabularies/'.format(
+                repo_slug=self.repo.slug,
+            ))
```

Comment:
why don't you use your function `get_vocabularies(self, repo_slug, expected_status=OK)` ?